pkg/actuators/machine/vm.go

```diff
@@ -128,16 +129,36 @@ func validateClusterConfig(mscp *machineScope, fldPath *field.Path) *field.Error
 			return field.Required(fldPath.Child("cluster", "uuid"), "Missing cluster uuid")
 		} else {
 			clusterUUID := *mscp.providerSpecValidated.Cluster.UUID
-			if _, err = mscp.nutanixClient.V3.GetCluster(clusterUUID); err != nil {
+			if res, err := mscp.nutanixClient.V3.GetCluster(clusterUUID); err != nil {
 				errMsg = fmt.Sprintf("Failed to find cluster with uuid %v. error: %v", clusterUUID, err)
 				return field.Invalid(fldPath.Child("cluster", "uuid"), clusterUUID, errMsg)
+			} else {
+				clusterName = *res.Spec.Name
 			}
 		}
 	default:
-		errMsg = fmt.Sprintf("Invalid cluster identifier type, valid types are: %q, %q.", machinev1.NutanixIdentifierName, machinev1.NutanixIdentifierUUID)
+		errMsg = fmt.Sprintf("Invalid cluster identifier type, valid types are: %q, %q.", configv1.NutanixIdentifierName, configv1.NutanixIdentifierUUID)
 		return field.Invalid(fldPath.Child("cluster", "type"), mscp.providerSpecValidated.Cluster.Type, errMsg)
 	}
 
+	// If the providerSpec configures the failureDomain reference, validate that the cluster
+	// configuration in both the providerSpec and the referenced failureDomain are consistent.
+	if mscp.failureDomain != nil {
```

Comment:
When is this validation run? Is it only on create? If the failure domains change over time, this might break updates to Machines if it requires the machine to be reconfigured

I think we need to assume Machines are immutable once created, even if the failure domain information they refer to changes

Reply:
Yes, the validation is only when creating the Machine VM. After the VM is created, any changes of the failure domains won't affect the existing Machines, but will affect the MachineSet when it creating new Machine instances with failureDomain reference configured. This is the same as the other fields of the Machine provider spec.